android-emulator-plugin/src/main/java/com/quittle/androidemulator/task/ProcessDestroyer.java

```diff
@@ -22,6 +24,7 @@
     @Override
     public Process apply(final Process process) {
         if (process != null) {
+            List<ProcessHandle> descendants = process.descendants().collect(Collectors.toList());
```

Comment:
I'm happy with this change, but why not just grab these and `forEach` the stream when you go to destroy them, rather than doing it upfront?

I suspect Java is just holding onto the PIDs here and if you kill ones that were previously killed, there's a minor possibility that you kill some other, new process that had it's PID recycled.